examples/offline_inference.py

```diff
@@ -1,22 +1,106 @@
+from dataclasses import asdict
+
+from transformers import AutoTokenizer
+
 from vllm import LLM, SamplingParams
+from vllm.engine.arg_utils import EngineArgs
+from vllm.utils import FlexibleArgumentParser
+
+
+def get_prompts(args):
+    # The default sample prompts.
+    prompts = [
+        "Hello, my name is",
+        "The president of the United States is",
+        "The capital of France is",
+        "The future of AI is",
+    ]
+
+    # if user specifies input-len, generate fake fixed-length prompts
+    # The code is copied from benchmark_throughput.py
+    if args.input_len is not None:
+        tokenizer = AutoTokenizer.from_pretrained(
+            args.tokenizer, trust_remote_code=args.trust_remote_code)
+
+        for i in range(-10, 10):
+            prompt = "hi " * (args.input_len + i)
+            tokenized_prompt = tokenizer(prompt).input_ids
+            if len(tokenized_prompt) == args.input_len:
+                break
+        else:
+            raise ValueError(
+                f"Failed to synthesize a prompt with {args.input_len} tokens.")
+
+        prompts = [prompt for _ in range(args.batch_size)]
```

Comment:
Thank you for the review. What you said is correct; we shouldn't use random prompts in the examples. However, I haven't come up with a quick way to generate prompts of a fixed length, so I removed the input length for simplicity. Since we're using `EngineArgs`, all options in the LLM are supported, and users can pass them directly through the command line. By running `python offline_inference.py -h`, they can see all the parameters. Therefore, I didn't explicitly add them to the script. Let me know if this approach works.